addons/mail/static/src/core/common/composer.js

```diff
@@ -430,10 +430,10 @@ export class Composer extends Component {
                         }
                     }),
                 };
-            case "Thread":
+            case "Channel":
```

Comment:
I would use `"discuss.channel"`, so it matches the model name and it's more idiomatic code.
The rest should also do it, although slightly out-of-scope of current PR.